pkg/provisioner/demo/demo.go

```diff
@@ -250,38 +248,13 @@ func (p *demoProvisioner) Provision(_ provisioner.ProvisionData, _ bool) (result
 // multiple times, and should return true for its dirty flag until the
 // deprovisioning operation is completed.
 func (p *demoProvisioner) Deprovision(_ bool) (result provisioner.Result, err error) {
-	hostName := p.objectMeta.Name
-	switch hostName {
-	default:
-		return result, nil
-	}
-
-	// p.log.Info("ensuring host is removed")
-
-	// result.RequeueAfter = deprovisionRequeueDelay
+	p.log.Info("deprovisioning host")
+	return result, nil
```

Comment:
I was not sure how to get rid of this `default switch-case only` issue, hence I am putting a log message, open for discussion.

Reply:
I don't really get why was this a switch case to  begin with...

Reply:
Is there any significance of accessing the `p.objectMeta.Name` field, could that have some effect? (I think no but maybe there is some GO magic that I am no considering)

Reply:
me neither and I am not sure if theres any go magic.

Reply:
This is for the "demo" provisioner, so it was probably meant as a way for developers to play with it. You could add specific cases for what you wanted to happen to the hosts. This is also why there is commented out code below. Just a simple way to make it do different things.
That said, I think we can safely remove it, otherwise it will just rot

Reply:
Yeah, I think it was just copy-and-paste.
(The purpose of the demo provisioner was to test a web UI, by allowing you to create a host that would sit in a certain state. The state it ended up in was determined by the host name.)